### PR TITLE
Token blacklist

### PR DESCRIPTION
**Pull request by paOol**
store local data on hidden tokens.

### CODE REVIEW

File: data/settings/actions.ts
Comment (SpicyPete) [inline data/settings/actions.ts]: Minor:  Change all instances of `tokenID` to `tokenId` to match conventions throughout the rest of the project

**Comment by paOol** [comment]
https://github.com/Bitcoin-com/badger-mobile/issues/173

File: data/store.ts
Comment (SpicyPete) [inline data/store.ts]: Persist config not required, instead add `settings` to the higher-level persistConfig on line `64`

File: screens/HomeScreen.tsx
Comment (SpicyPete) [inline screens/HomeScreen.tsx]: 1. `isRevealed` is not descriptive, what is revealed?  On the home-screen multiple things may be revealable later on.  Rename to more descriptive, such as `isHiddenTokensRevealed` or `isShowingBlacklist`, etc etc.  

2.  Follow the useState hook naming convention where the `setter` method is named as `setVariableName`.  In this case, that would be `setIsRevealed` (changing depending on how the variable name changes)

File: screens/HomeScreen.tsx
Comment (SpicyPete) [inline screens/HomeScreen.tsx]: Unnecessary function

File: screens/HomeScreen.tsx
Comment (SpicyPete) [inline screens/HomeScreen.tsx]: Adding an extra object assignment here, then passing it immediately.  
Either remove this extra assignment, or give it a meaningful name; `obj` means nothing.

File: screens/HomeScreen.tsx
Comment (SpicyPete) [inline screens/HomeScreen.tsx]: This `memo` value relies on `tokenBlacklist`, so that must be added as a dependency here, same with the curatedTokenData below.

File: screens/HomeScreen.tsx
Comment (SpicyPete) [inline screens/HomeScreen.tsx]: Extra complexity here.  RIght now it's computing the blacklistToken data, then listening to changes to this, to compute the `blacklistSection`.  
Since this first `memo` value is only used to directly trigger and modify the other `memo` value, it makes sense to combine the code into a single memo.  This way the related logic stays together, less looking around.    The primary goal of the hook paradigm is colocating related code by hook/effect blocks instead of lifecycle events.
For example, that may be done like

```javascript
const blacklistSection:  WalletSection = useMemo(() => {
  const filteredTokens = tokenData.filter(data => tokenBlacklist.include(data.tokenId);
   return {
    title: 'Hidden Tokens',
   data: filteredTokens
  };
}, [tokenData, tokenBlacklist])
```

Then same for `sectionSLP`

File: screens/HomeScreen.tsx
Comment (SpicyPete) [inline screens/HomeScreen.tsx]: See comment above, this memo can be removed

File: screens/HomeScreen.tsx
Comment (SpicyPete) [inline screens/HomeScreen.tsx]: Avoid direct string manipulation like this, prefer replacing the whole word rather than adding a single letter.  
eg:
`Hidden {tokenBlacklist.length > 1 ? "Tokens" : "Token"}`
This is to reduce the cost of adding internationalization later on, it's much easier to swap entire strings by language.  Not an issue right now as we don't have internationalization, but avoiding this pattern now will make it easier when we do.

File: screens/WalletDetailScreen.tsx
Comment (SpicyPete) [inline screens/WalletDetailScreen.tsx]: ?
What's the purpose of this effect

File: screens/WalletDetailScreen.tsx
Comment (SpicyPete) [inline screens/WalletDetailScreen.tsx]: `Blacklisted`, single word let's not capitalize it in the middle

File: screens/WalletDetailScreen.tsx
Comment (SpicyPete) [inline screens/WalletDetailScreen.tsx]: Styling / placement of this toggle needs work.

The top of the screen is the most valuable real-estate, whereas this feature is pretty low priority the majority of the time.  
Sync with me for ideas, lots to consider where/how it's placed, especially in regards to target areas, proximity to other things, etc.

Comment (SpicyPete) [inline screens/WalletDetailScreen.tsx]: All `Component` names should be `Uppercase`, to follow project conventions

File: screens/WalletDetailScreen.tsx
Comment (SpicyPete) [inline screens/WalletDetailScreen.tsx]: Will probably get changed anyways when we find a better button placement, but `this` can be omitted here as we're on this specific token's page already.  
Trying to think of how the messaging for this might be improved as the user doesn't know what show/hide/blacklists are.  Perhaps something which alludes to adding/removing from the token vault, since that's a concept/name we show to the user.

**Review by SpicyPete — CHANGES_REQUESTED** [review]

**Comment by SpicyPete** [comment]
Making great progress, just need to figure out the UI side of it,  added some questions / possible improvements to the code.

File: screens/HomeScreen.tsx
Comment (paOol) [reply screens/HomeScreen.tsx]: these changes caused the SectionList to crash. I was also getting Type errors when trying to refactor accordingly.

Seeing as how the functionality already works and there are no performance issues, I do not feel it is worth spending the extra cycles to refactor for little to no gain.

File: screens/WalletDetailScreen.tsx
Comment (paOol) [reply screens/WalletDetailScreen.tsx]: removed.

File: screens/HomeScreen.tsx
Comment (SpicyPete) [reply screens/HomeScreen.tsx]: Edit - After talking online, this is fine as is.

File: screens/HomeScreen.tsx
Comment (SpicyPete) [inline screens/HomeScreen.tsx]: Please follow the useState naming convention I outlined in last review, where the setter is based on the variable's name
So in this case it would be 
`const [isShowingBlacklist, setIsShowingBlacklist] = useState(false);`

See more: 
https://reactjs.org/docs/hooks-state.html

File: screens/HomeScreen.tsx
Comment (SpicyPete) [reply screens/HomeScreen.tsx]: The change above which makes this hook redundant is worth doing, as it places related logic closer to eachother.

**Review by SpicyPete — CHANGES_REQUESTED** [review]

File: screens/WalletDetailScreen.tsx
Comment (SpicyPete) [inline screens/WalletDetailScreen.tsx]: Call components as `jsx`  style 
eg. 
`<ShowButton tokenId={tokenId} />`

File: screens/WalletDetailScreen.tsx
Comment (SpicyPete) [inline screens/WalletDetailScreen.tsx]: `blackList` -> `blacklist`

File: screens/WalletDetailScreen.tsx
Comment (SpicyPete) [inline screens/WalletDetailScreen.tsx]: Not using `Text` in the app, please use the app specific `T` for consistent sizing

File: screens/WalletDetailScreen.tsx
Comment (SpicyPete) [inline screens/WalletDetailScreen.tsx]: Prefer multiples of 8 for spacing consistency.  `16` is used for left/right edge margins throughout the app, so in this case, use `16`.  
Purpose of the edge margins are for android phones with rounded edges to the screens.

File: screens/WalletDetailScreen.tsx
Comment (SpicyPete) [inline screens/WalletDetailScreen.tsx]: Use the `T` component, not the base `Text` component.  If we ever need to use a new font style, it should be implemented within `T`, or as an extension of T.

File: screens/WalletDetailScreen.tsx
Comment (SpicyPete) [inline screens/WalletDetailScreen.tsx]: This app already has many icon libraries available to use.
For example on how to use them, see 
https://github.com/Bitcoin-com/badger-mobile/blob/develop/components/CoinRow/CoinRow.tsx#L5

Here's a page with all of the icons already available
https://oblador.github.io/react-native-vector-icons/

The eye icons imported here IMO are a bit unclear as to what the closed one means, in the above page there are many more to choose from.  `FontAwesome5` and `Feather` both appear to have nice eyes with lines through icons.

File: screens/WalletDetailScreen.tsx
Comment (SpicyPete) [inline screens/WalletDetailScreen.tsx]: Right now the icon itself is unresponsive to press events.  

The common ways to handle this are outlined here
https://facebook.github.io/react-native/docs/handling-touches.html#touchables
This app is using `TouchableOpacity` in all cases right now, so I would recommend continuing that route for visual consistency.
As an added bonus TouchableOpacity will also bring visual feedback to the touch. 

Example used within app
https://github.com/Bitcoin-com/badger-mobile/blob/develop/components/CoinRow/CoinRow.tsx#L20

File: screens/WalletDetailScreen.tsx
Comment (SpicyPete) [inline screens/WalletDetailScreen.tsx]: Type names throughout the rest of the app are `UpperCased`

File: screens/WalletDetailScreen.tsx
Comment (SpicyPete) [inline screens/WalletDetailScreen.tsx]: Actions should be `verbs` which describe what will happen when pressed, but after looking at this and experimenting a bit, I think this may look best without any text at all.    
If it's confusing, then we could tie in the toggle to the available messaging box on the screen.  But that could happen later, for now I would try without any text and see if that looks good.

I do like the location here for this button, not pressing it at by mistake, even on my small phone.

**Review by SpicyPete — CHANGES_REQUESTED** [review]
Looking better.
Added some more comments

Main issues I'm seeing right now are
1.  No need to bring in new eye assets, use the extensive icon library which the app already uses.
2. the layout of the `text + icon` is visually odd, I would try it without the text since it looks weird sticking out to the left.  May also be able to find a good weighting by moving the text below/above and adjusting color, but no text is probably clear enough.  Can add more info if people ask about it.

File: screens/HomeScreen.tsx
Comment (SpicyPete) [inline screens/HomeScreen.tsx]: I think `Show` and `Hide` will be better verbs to use here.  Simpler words for non-native-English speakers.

File: screens/WalletDetailScreen.tsx
Comment (paOol) [reply screens/WalletDetailScreen.tsx]: resolved by using icon only.

**Review by SpicyPete — APPROVED** [review]
Looking and feeling great, approved for merge once #229 is merged into this branch.

File: screens/WalletDetailScreen.tsx
Comment (SpicyPete) [reply screens/WalletDetailScreen.tsx]: Touchable opacity is for the visual feedback, added this change in a PR.

**Pull request merged**
